sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReceiveLinkProcessor.java

```diff
@@ -218,12 +222,15 @@ public void onNext(ServiceBusReceiveLink next) {
             currentLink = next;
             next.setEmptyCreditListener(() -> 0);
 
+            unreceivedMessages.set(0);
```

Comment:
We set 0 here, the old link creadit will be lost, is it OK?

what I mean is the old logic is when we get a new link, we will calculate the credit by combining the link credit and pending message, but now we combine unreceived and pending messge.

Reply:
Here I reset the counter as we won't receive any messages from old link. (Including buffered messages from old link reactor-operator-chain).
I will test this edge case and add UT to cover.